angr/state_plugins/symbolic_memory.py

```diff
@@ -713,10 +715,12 @@ def _store(self, req):
         else:
             needed_bytes = self.state.se.eval(req.size)
             if needed_bytes < max_bytes:
-                sv = req.data[max_bytes*8-1:(max_bytes-needed_bytes)*8]
+                sv = req.data[max_bytes*self.state.arch.byte_width-1:
+                              (max_bytes-needed_bytes)*self.state.arch.byte_width]
```

Comment:
so it occurs to me that this stuff could be simplified by the ast.get_bytes() method from claripy, but that has a hardcoded 8 bit byte size. You would have to have a global setting in claripy for byte size, or maybe an extra param to that function?